keyboards/hub16/matrix.c

```diff
@@ -0,0 +1,270 @@
+/*
```

Comment:
This could probably be trimmed down quite a bit with Custom Matrix Lite (https://docs.qmk.fm/#/custom_matrix?id=_39lite39), but that can wait for a separate PR if you'd prefer.

Reply:
I might hold off for the time being, will look into it and submit a PR at a later date. Thanks!